dev-docs/2023-11-24 versioning ax.md

```diff
@@ -0,0 +1,106 @@
+# Versioning Ax and friends
+
+In this document there will be three main "players" (as of writing, the names are not
+fixed), I start by introducing them and their relevance to the project, then proceed
+to explain the considered solution.
+
+## "Players"
+
+The three main players are:
+- `ax` - the result of merging 3 tools:
+  - the previously existing `ax` CLI tool,
+    which handled several aspects of running an Actyx swarm;
+  - the `actyx` binary, which was the system that ran "as a node" in the swarm,
+    processing events, handling queries, etc;
+  - the `cert` tool, which was used for licensing purposes and complementing an Actyx
+    swarm's authorization and authentication capabilities.
+- `ax-core` - the library resulting of extracting the core functionality from the
+  `actyx` binary such that it can be exposed for other uses, such as being embedded
+  in an application
+- `node` - the node component from `actyx`, which can be thought of as the inner Actyx
```

Comment:
I’m not sold on using “node version”, since almost every programmer will think this refers to node.js. It also focuses on the swarm networking part, while the other interesting half is the event API and query language.

Thinking about options I only come up with generic terms like “service version”. We could also make it distinct and specific by calling it “AQL version”.

Reply:
Isn't this also encompassing aspects other than AQL that might change?
e.g. how node-to-node communicate, data retention behavior, etc?